web/src/components/tasks/environment-questionnaire/EnvQuestionnaireStepper.tsx

```diff
@@ -143,6 +144,31 @@ export const EnvQuestionnaireStepper = (): ReactElement => {
     }
   };
 
+  const handleOnStepClick = (newStep: number): void => {
+    const category = EnvPermitConfiguration.find((s) => newStep === s.stepIndex);
+
+    if (!category) return;
+
+    const categoryName = category.name.toLowerCase().split(" ").join("_");
+    analytics.event.gen_guidance_stepper_step_category.click.general_guidance_step(
+      newStep,
+      categoryName,
+    );
+
+    if (
+      envContext.state.stepIndex === 0 &&
+      isAuthenticated === IsAuthenticated.FALSE &&
+      userWantsToContinueWithoutSaving === false
+    ) {
+      updateQueue?.queuePreferences({ returnToLink: ROUTES.envPermit }).update();
+      setShowContinueWithoutSaving(true);
+      setShowNeedsAccountModal(true);
+      envContext.setStepIndex(newStep);
+      return;
+    }
+    envContext.setStepIndex(newStep);
+  };
```

Comment:
I know you didn't initially author this code, but I think this is a good opportunity for some cleanup and to leave things better than we found them. For that reason I'm going to categorize these as [dust], but I probably would have made them all [pebble]s if I reviewed this code the first go round.

Reply:
[dust]
I recommend pulling the condition out into a named variable that better articulates what we're actually checking for.
```suggestion

    const myConditionName = envContext.state.stepIndex === 0 && isAuthenticated === IsAuthenticated.FALSE && userWantsToContinueWithoutSaving === false;

    if (myConditionName) {
      updateQueue?.queuePreferences({ returnToLink: ROUTES.envPermit }).update();
      setShowContinueWithoutSaving(true);
      setShowNeedsAccountModal(true);
      envContext.setStepIndex(newStep);
      return;
    }
    envContext.setStepIndex(newStep);
  };
```

Reply:
[dust] It looks like both in both cases we're invoking `envContext.setStepIndex(newStep)`. You could simplify this slightly.

```suggestion
    if (
      envContext.state.stepIndex === 0 &&
      isAuthenticated === IsAuthenticated.FALSE &&
      userWantsToContinueWithoutSaving === false
    ) {
      updateQueue?.queuePreferences({ returnToLink: ROUTES.envPermit }).update();
      setShowContinueWithoutSaving(true);
      setShowNeedsAccountModal(true);
    }
    envContext.setStepIndex(newStep);
  };
```

Reply:
This is on me. I was going to refactor it when implementing the next version of the feature, but feel free to hack away at it now.